handlers/server_spec_handler.go

```diff
@@ -48,13 +47,14 @@ func (h *Handler) ServerVersionHandler(w http.ResponseWriter, r *http.Request) {
 
 	err = json.NewEncoder(w).Encode(version)
 	if err != nil {
-		logrus.Errorf("unable to send data: %v", err)
+		h.log.Error(ErrDataSend(err))
+		http.Error(w, ErrDataSend(err).Error(), http.StatusNotFound)
 	}
 }
 
 // CheckLatestVersion takes in the current server version compares it with the target
 // and returns the result (latest.CheckResponse)
-func CheckLatestVersion(serverVersion string) (*latest.CheckResponse, error) {
+func (h *Handler) CheckLatestVersion(serverVersion string) (*latest.CheckResponse, error) {
```

Comment:
Hey @dhruv0000 in this function @Chadha93 is adding the (h *Handler) so that he can use the h.log.Error(..). But this function is generating a problem in the `CheckLatestVersion` which is in the mesheryctl `version.go`. This is because the function `CheckLatestVersion` in mesheryctl does not have this h pointer.  And we cannot import the handler package in the mesheryctl. So me and @Chadha93 was thinking what could be the possible solution for this? You can find this function in mesheryctl version.go in **line number 125**

Reply:
The function is using logger to print the info for version it's returning. I think it will make more sense for it to be in `ServerVersionHandler` method and `checkMesheryctlClientVersion` function instead.